prerechnung.py

```diff
@@ -1,41 +1,89 @@
+from google.cloud import documentai_v1beta3 as documentai
+from google.api_core.client_options import ClientOptions
+
 
 def process(company_name):
-    prs = Api.api.get_list("PreRechnung",filters={'company':company_name,
-                                                  'processed':False},
-                           limit_page_length=LIMIT) 
+    prs = Api.api.get_list(
+        "PreRechnung",
+        filters={'company': company_name, 'processed': False},
+        limit_page_length=LIMIT
+    )
     for pr in prs:
         process_inv(pr)
     print("Prerechnungen vorprozessiert")
-    
+
+
+def extract_invoice_info(pdf_file_content) -> dict:
+    project_id = "quantum-idiom-379621"
+    processor_id = "b9f742465e96697"
+    location = "eu"
+    mime_type = "application/pdf"
+
+    # Instantiates a client
+    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = 'google-credentials.json'
+
+    # Set endpoint to EU
+    options = ClientOptions(api_endpoint="eu-documentai.googleapis.com")
+    # Instantiates a client
+    client = documentai.DocumentProcessorServiceClient(client_options=options)
+
+    # The full resource name of the processor, e.g.:
+    # projects/project-id/locations/location/processors/processor-id
+    name = f"projects/{project_id}/locations/{location}/processors/{processor_id}"
+
+    # Construct a document object
+    document = {"content": pdf_file_content, "mime_type": mime_type}
+
+    # Configure the process request
+    request = {"name": name, "document": document}
+
+    # Use the Document AI API to process the document
+    result = client.process_document(request=request)
+
+    print("Document processing complete.")
+
+    # Get all of the document text as one big string
+    document = result.document
+
+    # For a full list of Document object attributes, please reference this page:
+    # https://googleapis.dev/python/documentai/latest/_modules/google/cloud/documentai_v1beta3/types/document.html#Document
+    document_pages = document.pages
+
+    # Read the text recognition output from the processor
+    text = document.text
+
+    # Get the entities from the document
+    entities = []
+    for entity in document.entities:
+        props = []
+        for prop in entity.properties:
+            props.append({
+                "value": prop.normalized_value.text or prop.text_anchor.content or prop.mention_text,
+                "type": prop.type_,
+                "confidence": prop.confidence,
+            })
+        entities.append(
+            {
+                "value": entity.normalized_value.text or entity.text_anchor.content or entity.mention_text,
+                "type": entity.type_,
+                "properties": props,
+                "confidence": entity.confidence,
+            }
+        )
+    # Return the results as a dictionary
+    return {"document_text": text, "entities": entities}
+
+
 def process_inv(pr):
     print(pr['name'])
-    inv = purchase_invoice.PurchaseInvoice(pr['lager'])
```

Comment:
please keep all the old code for now, but execute it only if no Google credentials are available: `if '-google-credentials-' in sg.UserSettings() ...`